### PR TITLE
Travelling pointing arrows, brains can no longer point

### PR DESCRIPTION
**Pull request by Krunklehorn**
## About the PR

Makes pointing arrows travel from the pointer to the target and animates them smoothly.

Fixes brain pointing using a new attempt event.

## Why / Balance

This makes it much easier to tell who is pointing and at what.

## Technical details

On the client side, `PointingSystem` no longer reuses the visualizer for gravity. The verb-check has been converted to a new attempt event under `SharedPointingSystem`.

On the server side, entity coordinates local to the arrow are sent during `TryPoint()`.

In Shared, AnimationTime was moved out of the component to a readonly variable.

## Media

Initial:
vlc_sHpIgBDSHV

Reduced overshoot, a bit punchier:
vlc_RuGfs733rl

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: Krunk
- add: Pointing arrows now begin at the player and smoothly animate to their target.

### CODE REVIEW

**Comment by potato1234x** [comment]
the overshoot that they do looks a bit off to me

**Comment by Krunklehorn** [comment]
I'll tighten up the initial movement just a touch.

**Comment by Krunklehorn** [comment]
It's difficult to see in the gif but the second version is still quite smooth.

**Comment by UbaserB** [comment]
I had a dream about this last night about using this feature ingame. The sheer amount of "bounciness" the arrow had was very disorientating and worked weirdly. No joke though, for some reason i dreamed about it. Maybe it was on my mind, who knows.

**Comment by K-Dynamic** [comment]
I want to recreate this moment where a secoff gets bullied by the power of 15 different people pointing at them https://youtu.be/m_yIi7Q9NzQ?t=1m30s

File: Content.Client/Pointing/PointingSystem.cs [hunk 1]
```diff
@@ -49,6 +46,9 @@ private void AddPointingVerb(GetVerbsEvent<Verb> args)
         if (_mobState.IsIncapacitated(args.User))
             return;
 
+        if (HasComp<SleepingComponent>(args.User))
+            return;
+
```
Comment (metalgearsloth) [inline Content.Client/Pointing/PointingSystem.cs:51]: This should be raising an event / checking CanInteract rather than hardcoding in a bunch of systems.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/Silicons/Borgs/BorgSystem.cs [hunk 2]
```diff
@@ -242,6 +244,12 @@ private void OnBrainMindAdded(EntityUid uid, BorgBrainComponent component, MindA
         _mind.TransferTo(mindId, containerEnt, mind: mind);
     }
 
+    private void OnPointAttempt(EntityUid uid, BorgBrainComponent component, PointAttemptEvent args)
+    {
+        if (!HasComp<BorgChassisComponent>(uid))
```
Comment (deltanedas) [inline Content.Server/Silicons/Borgs/BorgSystem.cs:249]: 99% sure this wont work

Comment (Krunklehorn) [reply Content.Server/Silicons/Borgs/BorgSystem.cs:249]: I just tested it. PB couldn't point until I put it inside a chassis. Lost the ability when I took it out again.

Do we expect it to break anything else?

Comment (deltanedas) [reply Content.Server/Silicons/Borgs/BorgSystem.cs:249]: the posibrain isnt doing the pointing the borg is

Comment (deltanedas) [reply Content.Server/Silicons/Borgs/BorgSystem.cs:249]: when in the chassis the minds OwnedEntity is the borg chassis, not the brain
BorgChassis is never added to the brain

shrimply just have borgbrain always cancel it like meat brain

Comment (Krunklehorn) [reply Content.Server/Silicons/Borgs/BorgSystem.cs:249]: Ohhh I see, the check isn't needed then.

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**